airflow/contrib/operators/qubole_operator.py

```diff
@@ -240,3 +241,9 @@ def __setattr__(self, name, value):
             self.kwargs[name] = value
         else:
             object.__setattr__(self, name, value)
+
+    @classmethod
+    def get_serialized_fields(cls):
+        """Serialized QuboleOperator contain exactly these fields."""
+        cls._serialized_fields = frozenset(super().get_serialized_fields() | {"qubole_conn_id"})
```

Comment:
To pull across the comments from the closed pr: this should have the `if not cls._serialized_fields` guard here, otherwise we re-calculate this each time it's called (not the end of the world, but still)

Reply:
The problem is when calling `super().get_serialized_fields()` it would set `cls._serialized_fields` from the BaseOperator, hence it would not have `{"sql"}` in it

Reply:
Oh yeah.

There's no point storing it on the class variable then is there? Just `return super().get_serialized_fields() | frozenset(["quoble_conn_id"]}`?

Reply:
The only reason I stored it is for consistency as otherwise `cls._serialized_fields` will have a different value than `cls.get_serialized_fields()`. We don't use `cls._serialized_fields` anywhere, maybe we can just remove it from the Base Class too than WDYT? If not, let's keep it at both places as the results won't match

Reply:
Oh hang on - I've got an idea. So if you delcare a var called `__x` python automatically converts that to `_MyClass__x`, so we can do this:

```python
class Base:
    def __init__(self, x=1):
        self.x = x

    __serialized_fields = None

    @classmethod
    def get_serialized_fields(cls):
        if not cls.__serialized_fields:
            cls.__serialized_fields = frozenset(vars(Base()))
        return cls.__serialized_fields


class  Sub(Base):
    def __init__(self, y, **kwargs):
        super().__init__(**kwargs)
        self.y = y

    __serialized_fields = None
    @classmethod
    def get_serialized_fields(cls):
        if not cls.__serialized_fields:
            cls.__serialized_fields = super().get_serialized_fields() | frozenset({'y'})
        return cls.__serialized_fields


print(Sub.get_serialized_fields())
print(Base.get_serialized_fields())
```

And then get as output

```
frozenset({'x', 'y'})
frozenset({'x'})
```

https://docs.python.org/3/tutorial/classes.html?highlight=class%20private#tut-private

Reply:
Applied

Reply:
Didn't know that this kind of thing existed :D 

Thanks @ashb